addons/stock/models/procurement.py

```diff
@@ -179,14 +243,14 @@ def run(self, product_id, product_qty, product_uom, location_id, name, origin, v
         values.setdefault('priority', '1')
         values.setdefault('date_planned', fields.Datetime.now())
         rule = self._get_rule(product_id, location_id, values)
-
```

Comment:
- think about a solution to call run(push) or run(pull)
- look at search_rule and push_apply duplication

Reply:
No mechanism for grouping , propagate warehouse for push rule before , are you want to implement it ? I prefer to make new one with business cases